draft-spaghetti-grow-nrtm-v4.xml

```diff
@@ -570,9 +584,10 @@
     <section title="Purpose">
       <t>
         A Delta File contains all changes for exactly one incremental update of the IRR Database.
-        It may include new, modified and deleted objects. Delta Files can contain multiple updates to multiple objects.
+        It may include new, modified and deleted objects. Delta Files can contain multiple alternations to multiple objects.
       </t>
     </section>
+    <!--SK: Under which circumstances a new Delta file is published? Is this something that we need to define? -->
```

Comment:
Does https://htmlpreview.github.io/?https://github.com/mxsasha/nrtmv4/blob/main/draft-spaghetti-grow-nrtm-v4.html#name-delta-files-2 not suffice for that? About once a minute, if there were changes, with catching up permitted after e.g. downtime.

Reply:
Hi Sasha, yes it is quite sufficient. But upon reviewing the document I goth the feeling that is not a fixed "per-minute" timeframe but can be "up-to-a-minute", which means we allow some room for earlier/faster delta files. Is that correct or I misunderstood it? Also in the same first bullet of 3.3.1. says " if there have been changes to RPSL objects in that time frame". Question is what if there are no changes? We skip the publication of Delta files or we publish a delta file with zero changes, regardless ? If we skip the publication, then the next one is published in the +2 or +3 minute ? 

I am not sure If I described my question correctly, if not I am happy to elaborate again.

Reply:
I see. What I had in mind is this: every minute, a mirror server looks up what was included in the last delta file it published. If there have been changes to RPSL objects between "now" and that last file, produce a delta file.

This means:
* No changes, no delta file. We would just clutter the repo.
* The one minute publication timeframe does not have to be exact to the second. But not 5 minutes either. The idea is that this keeps the mirroring delay to one minute on average.
* If somehow the delta producer did not run, e.g. because that process was shut down, it will produce one bigger delta file upon the next run (if there were changes in the mean time).

Regarding timing, I'm thinking you might have:
```
10:15:10: Delta publisher runs, finds no changes, stops
10:15:11: An RPSL object is changed
10:16:10: Delta publisher runs, finds one change and publishes it, stops
10:17:10: Delta publisher runs, finds no changes, stops
10:18:10: Delta publisher runs, finds no changes, stops
10:18:53: An RPSL object is changed
10:19:10: Delta publisher runs, finds two changes and publishes them, stops
10:20:10: Delta publisher runs, finds no changes, stops
```